src/jit/lowerxarch.cpp

```diff
@@ -2096,34 +2186,37 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTree* tree)
             info->addInternalCandidates(l, l->internalFloatRegCandidates());
         }
 
-        if (haveLocalAddr)
-        {
-            MakeSrcContained(putArgStkTree, srcAddr);
-        }
-
         // If src or dst are on stack, we don't have to generate the address into a register
         // because it's just some constant+SP
         putArgStkTree->gtPutArgStkKind = GenTreePutArgStk::PutArgStkKindUnroll;
     }
+#ifdef _TARGET_X86_
+    else if (putArgStkTree->gtNumberReferenceSlots != 0)
+    {
+        putArgStkTree->gtPutArgStkKind = GenTreePutArgStk::PutArgStkKindRepInstr;
```

Comment:
> I'm confused why you would set the kind this way. We never use rep movs for a struct with GC slots, correct?

That's right. It wasn't entirely clear what the invariants were regarding `gtPutArgStkKind` when I added this case. I will change this.